### DISCUSSION TITLE
filter function not working tho the code looks correct?

### DISCUSSION THREAD

**Discussion by SabSchek** [neuefische/questions#254] [Web] [answered]
Hello,

I have a question and hope someone could give me a helpful advise why this it doesn't work. 
Earlier I changed some things in my styling when I saw later that the filter function isn't running anymore. 

See attached a couple of  pictures of the code:

<img width="1386" alt="Bildschirmfoto 2022-10-12 um 21 14 58" src="https://user-images.githubusercontent.com/109827469/195429514-bbcf53e9-034d-4014-b603-7aa285602188.png">
<img width="1440" alt="Bildschirmfoto 2022-10-12 um 21 13 58" src="https://user-images.githubusercontent.com/109827469/195429535-0ab477a5-677a-4585-8355-3fa1451d7c78.png">

Thank you in advance

**Comment by NikRoe** [comment]
Hi Fine,
please send us a link a PR, so you can take a closer look.

Thanks :)

**Reply by NikRoe** [reply]
https://github.com/SabSchek/capstone-project/pull/15

**Comment by NikRoe** [comment, marked as answer]
Hey Fine,

in your `[name].js` you work with a styled component called `AppBody`. However, it is not declared in the page itself -> you need to create the styled component in this page.
The filter itself works just fine. For instance, if you follow this [link](http://localhost:3000/category/Kleidung) while your server is running, the page is displayed properly. (You need to fix the issue with the `AppBody` first, of course)

EDIT: We noticed that one category does not match (Rad vs. Raeder). Make sure that you name your categories consistently.

EDIT2.0: Some of the imported Objects in the `goods`-array had the same `id`. This caused some issues when mapping over them, since they had the `id` as keys.

Best,
-Niklas
